hangman_game.py

```diff
@@ -0,0 +1,161 @@
+            print('  |      ')
+            print('  |      ')
+            print('__|__\n')
+        if self.hangmanState == 5:
+            print('  |‾‾‾‾|')
+            print('  |    0')
+            print('  |      ')
+            print('  |      ')
+            print('__|__\n')
+        if self.hangmanState == 4:
+            print('  |‾‾‾‾|')
+            print('  |    0')
+            print('  |   / ')
+            print('  |      ')
+            print('__|__\n')
+        if self.hangmanState == 3:
+            print('  |‾‾‾‾|')
+            print('  |    0')
+            print('  |   /| ')
+            print('  |      ')
+            print('__|__\n')
+        if self.hangmanState == 2:
+            print('  |‾‾‾‾|')
+            print('  |    0')
+            print('  |   /|\ ')
+            print('  |      ')
+            print('__|__\n')
+        if self.hangmanState == 1:
+            print('  |‾‾‾‾|')
+            print('  |    0')
+            print("  |   /|\ ")
+            print('  |   / ' )
+            print('__|__\n')
+        if self.hangmanState <= 0:
+            print('  |‾‾‾‾|')
+            print('  |    0')
+            print("  |   /|\ ")
+            print("  |   / \ ")
+            print('__|__\n')
+
+    def showSplit(self, array):
+        print(' '.join(array))  # format output for hangman
+
+    def playGame(self, minLength):
+        import numpy as np
+        target = self.setupGame(minLength)  # specify target word with setupGame
+        targetSplit = np.array(list(target))  # created for finding correct guesses
+        maskedSplit = np.array(['_' for letter in list(target)])  # created to show state of hangman game
+
+        while self.hangmanState > 0:
+            self.showBoard()  # start off by showing state of hangman
+            self.showSplit(maskedSplit)  # display state of board
+            # print(target)  # used for debugging
+            letterGuess = input('Guess a letter ')
+            if letterGuess.upper() in self.validLetters:
+                if letterGuess.upper() in self.guessedLetters:
+                    self.hangmanState -= 1  # decrement for repeated guess
+                    print('Already guessed this letter')
+                else:
+                    if letterGuess.upper() in targetSplit:
+                        validSpots = np.where(targetSplit == letterGuess.upper())  # all indices of match
+                        self.guessedLetters.append(letterGuess.upper())  # add guess to list of guesses
+                        for index in validSpots:
+                            maskedSplit[index] = letterGuess.upper()  # replace masked spot with letter
+                        if '_' not in maskedSplit:
+                            print('You won the game!')
+                            self.showSplit(maskedSplit)  # show the completed word
+                            self.showBoard()
+                            break  # exit the game since word was correctly guessed
```

Comment:
the use of break is ok. This is a fairly straight forward loop, howeve rwith complicated loops, it is sometimes dangerous to use break statements. for practice can we retool this to exit a bit more elegantly, maybe set the state variable to -1 or something.